src/lib.rs

```diff
@@ -5,100 +5,156 @@ pub struct DB<T> {
 
 /// An immutably borrowed subset of a DB
 ///
-/// NB: (nota bene, or "take special note"): You will need to be explcit about the liftimes in this
-/// struct
+/// NB: (nota bene, or "take special note"): You will need 
+/// to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBView<T> {
-    entries: Vec<&T>,
+pub struct DBView<'a, T:'a> {
+    entries: Vec<&'a T>,
 }
 
 /// An mutably borrowed subset of a DB
 ///
 /// NB: You will need to be explcit about the liftimes in this struct
 #[derive(Debug, PartialEq, Eq)]
-pub struct DBViewMut<T> {
-    entries: Vec<&mut T>,
+pub struct DBViewMut<'a, T:'a> {
+    entries: Vec<&'a mut T>,
 }
 
+// TODO change these functions to use iter filter and iter collect instead
 /// Filters a DBView using the the given predicate.
-pub fn filter_one<T, F>(view: &DBView<T>, predicate: F) -> DBView< T>
+pub fn filter_one<'a, 'b, T, F>(view: &'b DBView<'a, T>,
+                                predicate: F) -> DBView<'a, T>
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    let mut vec : Vec<&'a T> = Vec::new();
+    for elem in &view.entries {
+        if predicate(elem) {
+            vec.push(elem)
+        }
+    }
+    DBView{entries: vec}
 }
 
-/// Filters two DBView structs using the same predicate, producing two separate results. This is
-/// the moral equivalent of doing the two filters separately.
-pub fn filter_two<T, F>(view_a: &DBView<T>,
-                        view_b: &DBView<T>,
+/// Filters two DBView structs using the same predicate, producing two separate
+/// results. This is the moral equivalent of doing the two filters separately.
+pub fn filter_two<'a, 'b, 'c, T, F>(view_a: &'c DBView<'a, T>,
+                        view_b: &DBView<'b, T>,
                         predicate: F)
-                        -> (DBView<T>, DBView<T>)
+                        -> (DBView<'a, T>, DBView<'b, T>)
     where F: Fn(&T) -> bool
 {
-    unimplemented!()
+    let mut vec_a : Vec<&'a T> = Vec::new();
+    let mut vec_b : Vec<&'b T> = Vec::new();
+    for elem in &view_a.entries {
+        if predicate(elem) {
+            vec_a.push(elem)
+        }
+    }
+    for elem in &view_b.entries {
+        if predicate(elem) {
+            vec_b.push(elem)
+        }
+    }
+    (DBView{entries: vec_a}, DBView{entries: vec_b})
 }
```

Comment:
As you said, this could be cleaned up. Some good places to start: `Iterator::filter`, `Iterator::collect`, and `Iterator::cloned`.